src/core_modules/capture-core/components/Pages/MainPage/MainPage.container.js

```diff
@@ -120,6 +120,14 @@ const MainPageContainer = () => {
         showAllAccessible,
         categoryOptionIsInvalidForOrgUnit,
     });
+    const reduxSelectedTemplateId
+        = useSelector(({ workingListsTemplates }) => workingListsTemplates.teiList?.selectedTemplateId);
+    const workingListProgramId = useSelector(({ workingLists }) => workingLists
```

Comment:
I'm wondering if it's more sensible to get workingListProgramId from `workingListsContext.teiList?.programIdView`. I think the value is the same, it's more of a a matter of basing it on context rather than currentRequest. WDYT?

Reply:
Yes, makes sense!